### PR TITLE
Broker schema registry (1/3)

### PR DESCRIPTION
**Pull request by mgodave**
### Motivation

Allow users of the system to store the structure and format of the data in a topic

### Modifications

Add Schema information to protocol and empty interface to broker.

Two more dependent PRs:
https://github.com/mgodave/incubator-pulsar/pull/1
https://github.com/mgodave/incubator-pulsar/pull/2

### CODE REVIEW

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java [hunk 1]
```diff
@@ -422,52 +428,59 @@ protected void handleSubscribe(final CommandSubscribe subscribe) {
...
-                                    log.debug(
-                                            "[{}][{}][{}] Failed to create consumer because exclusive consumer is already connected: {}",
-                                            remoteAddress, topicName, subscriptionName,
-                                            exception.getCause().getMessage());
-                                }
-                            } else {
-                                log.warn("[{}][{}][{}] Failed to create consumer: {}", remoteAddress, topicName,
-                                        subscriptionName, exception.getCause().getMessage(), exception);
-                            }
+                service.getTopic(topicName).thenCompose(topic ->
```
Comment (sijie) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java]: Can I confirm that the change here is only adding a `thenApply` to convert `Consumer` to `TopicConsumer`?

**Comment by sijie** [comment]
@mgodave the ci fails because new files don't have correct license headers. 

you can run "mvn license:format" to format those new files with correct license headers.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java (hunk 1 above)
Comment (mgodave) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java]: Yes. Looks like the formatting changed because of it.

**Comment by mgodave** [comment]
@sijie Headers added and build passing.

**Comment by mgodave** [comment]
Replaced with https://github.com/apache/incubator-pulsar/pull/1137

**Pull request closed without merging**